LibraBFT/Impl/Consensus/SafetyRules/Properties/SafetyRules.agda

```diff
@@ -337,12 +339,16 @@ module constructAndSignVoteMSpec where
           (λ r → logInfo >> pure r) (Contract pre epoch round)
           x post (LogInfo fakeInfo ∷ outs)
 
-      convNoOuts : ∀ {outs} → NoMsgOuts outs  → NoMsgOuts (LogInfo fakeInfo ∷ outs ++ LogInfo fakeInfo ∷ [])
-      convNoOuts{outs} pf rewrite List-filter-++ isOutputMsg? outs (LogInfo fakeInfo ∷ []) | pf = refl
-
       pf : ∀ r st outs → Contract pre epoch round r st outs → Post r st outs
       pf r st outs (mkContract noOuts inv resultCorrect) .r refl .unit refl =
-        mkContract (convNoOuts{outs} noOuts) inv resultCorrect
+        mkContract (++-NoMsgOuts outs (LogInfo fakeInfo ∷ []) noOuts refl) inv resultCorrect
```

Comment:
This looks much nicer than the similar-but-uglier effort I've made for fixing continue1.contract to tolerate logging :smile:.  I really must learn to review and remember your work before doing anything myself!